include/shell.h

```diff
@@ -52,24 +52,26 @@ class BatchFile {
 	Bit16u file_handle;
 	Bit32u location;
 	bool echo;
-	DOS_Shell * shell;
-	BatchFile * prev;
-	CommandLine * cmd;
-	std::string filename;
+	DOS_Shell *shell = nullptr;
+	BatchFile *prev = nullptr;
+	CommandLine *cmd = nullptr;
+	std::string filename{};
 };
 
 class AutoexecEditor;
 class DOS_Shell : public Program {
 private:
 	friend class AutoexecEditor;
-	std::list<std::string> l_history, l_completion;
+	std::list<std::string> l_history{};
+	std::list<std::string> l_completion{};
+
+	char *completion_start = nullptr;
+	Bit16u completion_index = 0;
 
-	char *completion_start;
-	Bit16u completion_index;
-	
 public:
 
 	DOS_Shell();
+	~DOS_Shell();
```

Comment:
Add `override` keyword, please.

I think it will require adding `override` keyword to method `DOS_Shell::Run` (to avoid new warnings) - it's only one function, so it can be done now :)